core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -2703,6 +2711,26 @@ class ReplicaManagerTest {
       assertEquals(HostedPartition.None, replicaManager.getPartition(tp0))
       assertFalse(readRecoveryPointCheckpoint().contains(tp0))
       assertFalse(readLogStartOffsetCheckpoint().contains(tp0))
+
+      // verify that there is log to be deleted
+      val logsToBeDeleted = replicaManager.logManager.logsToBeDeleted
```

Comment:
This seems an indirect way of testing the logic. logsToBeDeleted will be picked up by the scheduler in LogManager, which uses mockTime. Could we just advance the mock time to trigger the scheduler?

Reply:
The call hierarchy for the method deleteLogs is
LogManager.deleteLogs()  (kafka.log)
LogManager.startupWithConfigOverrides(LogConfig, Map<String, LogConfig>)  (kafka.log)
LogManager.startup(Set<String>)  (kafka.log)

Currently, the tests in this file didn't trigger LogManager.startup, and therefore the scheduler is effectively not used. I feel the way of testing the background deletion in this PR is actually more direct for capturing the KafkaStorageException. Please let me know if I'm missing something.

Reply:
Thanks for the explanation. Sounds good.